Template/Trigger/ElementVisibilityTrigger.web.js

```diff
@@ -272,47 +257,148 @@
                         return;
                     }
                     var percentVisible = Math.max(getPercentVisible(entry.target), minPercentVisible);
+                    commonTrigger(triggerEvent, percentVisible, entry.target);
+                    commonTriggeredNodeCheck(entry.target);
+                }
+            });
+        }
 
-                    triggerEvent({
-                        event: 'mtm.ElementVisibility',
-                        'mtm.elementVisibilityPercentage': Math.round(percentVisible * 100) / 100,
-                        'mtm.elementVisibilityId': dom.getElementAttribute(entry.target, 'id'),
-                        'mtm.elementVisibilityClasses': dom.getElementClassNames(entry.target),
-                        'mtm.elementVisibilityText': TagManager.utils.trim(entry.target.innerText),
-                        'mtm.elementVisibilityNodeName': entry.target.nodeName,
-                        'mtm.elementVisibilityUrl': entry.target.href || dom.getElementAttribute(entry.target, 'href'),
-                    });
+        function isNodeEventTriggered(node) {
+            for (var j = 0; j < triggeredNodes.length; j++) {
+                if (node === triggeredNodes[j]) {
+                    return true;
+                }
+            }
 
-                    if (fireTriggerWhen === 'oncePage') {
-                        blockTrigger = true;
-                        TagManager.dom.bySelector(selectors).forEach(function (element) {
-                            observerIntersection.unobserve(element);
-                            triggeredNodes.push(element);
-                        });
-                        if (self.scrollIndex) {
-                            TagManager.window.offScroll(self.scrollIndex);
+            return false;
+        }
+
+        function setMutationObserver(triggerEvent) {
+            return function () {
+                if (observeDomChanges && isMutationObserverSupported) {
+                    notInDomNodes = getDynamicNodes(selectors);
+                    var targetNode = documentAlias.body;
+                    var config = {attributes: true, childList: true, subtree: true};
+                    observerMutation = new MutationObserver(function (mutationsList) {
+                        if (mutationObseverTimeout) {
+                            return;
                         }
-                    } else if (onlyOncePerElement) {
-                        observerIntersection.unobserve(entry.target);
-                        triggeredNodes.push(entry.target); // to avoid possible memory leaks as much as possible we add onceElement only when needed
+
+                        mutationObseverTimeout = true;
+                        setTimeout(function () {
+                            mutationObseverTimeout = false;
+                            mutationObserverCallback(mutationsList, triggerEvent);
+                            notInDomNodes = getDynamicNodes(selectors);
```

Comment:
not sure we can do this actually since a selector might match at a later point?

Reply:
Yes, removed